app/api/discount_codes.py

```diff
@@ -115,9 +143,54 @@ class DiscountCodeDetail(ResourceDetail):
     Discount Code detail by id or code.
     """
 
+    def decide_schema(self, json_data):
+        """To decide discount code schema based on posted data.
+        :param json_data:
+        :return:"""
+
+        used_for = json_data['data']['attributes'].get('used-for')
+        if not used_for:
+            try:
+                discount = self.session.query(DiscountCode).filter_by(id=int(json_data['data']['id'])).one()
+            except NoResultFound:
+                raise ObjectNotFound({'parameter': '{id}'}, "DiscountCode: not found")
+            used_for = discount.used_for
```

Comment:
If I understood this correctly what you're trying to do is disallow updating the value of `used_for`. Right ?? 
If yes, then kindly raise a `ConflictException` rather than silently changing back to it's previous value.